src/data-model-providers/codegen/CodegenDataModelProvider.cpp

```diff
@@ -161,6 +163,20 @@ CHIP_ERROR CodegenDataModelProvider::Startup(DataModel::InteractionModelContext
 
     InitDataModelForTesting();
 
+    // If no ConfigurationVersion was ever set, initialize it to 1.
+    chip::StorageKeyName kStorageKey = chip::DefaultStorageKeyAllocator::ConfigurationVersion();
+    if (!mPersistentStorageDelegate->SyncDoesKeyExist(kStorageKey.KeyName()))
+    {
+        mConfigurationVersion = 1;
+        uint16_t size         = sizeof(mConfigurationVersion);
+        mPersistentStorageDelegate->SyncSetKeyValue(kStorageKey.KeyName(), &mConfigurationVersion, size);
+    }
+    else
+    {
+        uint16_t size = sizeof(mConfigurationVersion);
+        mPersistentStorageDelegate->SyncGetKeyValue(kStorageKey.KeyName(), &mConfigurationVersion, size);
```

Comment:
these return a CHIP_ERROR. we should handle that.

Could we always "Get" and if we get a not found error then do the "set to 1 and save" ?

Reply:
Given this is an else on a check for SyncDoesKeyExist, I would assume it should never return a not found error, but sure. Let me add it just in case.

Reply:
Addressed in latest commit